web-core/src/main/java/com/tutrit/gateway/CarGateway.java

```diff
@@ -0,0 +1,10 @@
+package com.tutrit.gateway;
+
+import com.tutrit.bean.Car;
+
+import java.util.Optional;
+
+public interface CarGateway {
+    Optional<Car> saveCar(Car car);
```

Comment:
save car will  definitely return not null object, so not a place to use Optional.

Reply:
I got it. It's changed